dldashboard/web/src/components/Memory2D.js

```diff
@@ -102,8 +102,17 @@ class Memory2D extends React.Component {
     bot_y = height - bot_y;
 
     let renderedObjects = [];
+    let mapBoundry = [];
     let j = 0;
 
+    // Visualize map
+    obstacle_map.forEach((obj) => {
```

Comment:
nice! Does the entire `mapBoundary` get rebuilt for every frame?

Reply:
Yes, it gets rebuild every frame.